prime-router/src/main/kotlin/azure/ValidateFunction.kt

```diff
@@ -0,0 +1,328 @@
+
+                    // return CREATED status, report submission was successful
+                    HttpStatus.CREATED
+                } else HttpStatus.OK
+            } catch (e: ActionError) {
+                actionHistory.trackLogs(e.details)
+                HttpStatus.BAD_REQUEST
+            } catch (e: IllegalArgumentException) {
+                actionHistory.trackLogs(
+                    ActionLog(InvalidReportMessage(e.message ?: "Invalid request."), type = ActionLogLevel.error)
+                )
+                HttpStatus.BAD_REQUEST
+            } catch (e: IllegalStateException) {
+                actionHistory.trackLogs(
+                    ActionLog(InvalidReportMessage(e.message ?: "Invalid request."), type = ActionLogLevel.error)
+                )
+                HttpStatus.BAD_REQUEST
+            }
+
+        val submission = DetailedSubmissionHistory(
+            1,
+            TaskAction.none,
+            OffsetDateTime.now(),
+            httpStatus.value(),
+            mutableListOf<DetailedReport>(),
+            actionHistory.actionLogs.map {
+                DetailedActionLog(
+                    it.scope,
+                    it.reportId,
+                    it.index,
+                    it.trackingId,
+                    it.type,
+                    it.detail
+                )
+            }
+        )
+
+        // set status for validation response
+        submission.overallStatus = if (httpStatus == HttpStatus.BAD_REQUEST)
+            DetailedSubmissionHistory.Status.ERROR
+        else
+            DetailedSubmissionHistory.Status.VALID
+
+        return request.createResponseBuilder(httpStatus)
+            .header(HttpHeaders.CONTENT_TYPE, "application/json")
+            .body(
+                JacksonMapperUtilities.allowUnknownsMapper
+                    .writeValueAsString(submission)
+            )
+            .build()
+    }
+
+    /**
+     * Extract client header from request headers or query string parameters
+     * @param request the http request message from the client
+     */
+    private fun extractClient(request: HttpRequestMessage<String?>): String {
+        // client can be in the header or in the url parameters:
+        return request.headers[CLIENT_PARAMETER]
+            ?: request.queryParameters.getOrDefault(CLIENT_PARAMETER, "")
+    }
+
+    /**
+     * Extract the optional payloadName (aka sender-supplied filename) from request headers or query string parameters
+     * @param request the http request message from the client
+     */
+    private fun extractPayloadName(request: HttpRequestMessage<String?>): String? {
+        // payloadName can be in the header or in the url parameters.  Return null if not found.
+        return request.headers[PAYLOAD_NAME_PARAMETER]
+            ?: request.queryParameters[PAYLOAD_NAME_PARAMETER]
+    }
+
+    private fun processingType(request: HttpRequestMessage<String?>, sender: Sender): Sender.ProcessingType {
```

Comment:
The validate endpoint doesn't care about sync/async, as it never gets to the 'routing' part that uses processing mode. Remove this function entirely and just send in a hard coded 'false' for the async parameter where needed since you would not want this validate endpoint to run as async ever anyway.